influent-transport/src/main/java/influent/internal/util/Inet4Network.java

```diff
@@ -0,0 +1,54 @@
+package influent.internal.util;
+
+import java.math.BigInteger;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+public class Inet4Network implements InetNetwork {
+  private final static byte[] MAX = {
+      (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff,
+  };
+  private final static BigInteger MAX_VALUE = new BigInteger(1, MAX);
+  private InetAddress network;
+  private InetAddress netmask;
+
+  public Inet4Network(InetAddress base, int netmask) {
+    this.netmask = generateNetmask(netmask);
+    this.network = InetNetwork.maskIP(base, this.netmask);
+  }
+
+  public InetAddress getNetwork() {
+    return network;
+  }
+
+  public InetAddress getNetmask() {
+    return netmask;
+  }
+
+  @Override
+  public boolean contains(InetAddress address) {
+    try {
+      return network.equals(InetNetwork.maskIP(address, netmask));
+    } catch (IllegalArgumentException ex) {
+      return false;
+    }
+  }
+
+  private InetAddress generateNetmask(int mask) {
```

Comment:
How about using [java.util.BitSet](https://docs.oracle.com/javase/jp/8/docs/api/java/util/BitSet.html)?
It may make the logic simple.

```java
// prepare InetNetwork
BitSet subnetmask = new BitSet(32);
subnetmask.set(0, 24);
BitSet networkAddress = BitSet.valueOf(InetAddress.getByName("192.168.1.0").getAddress());
networkAddress.and(subnetmask);

// test client address
BitSet peerAddress = BitSet.valueOf(InetAddress.getByName("192.168.1.1").getAddress());
peerAddress.and(subnetmask);
peerAddress.equals(networkAddress)
```

Reply:
I've fixed using BitSet.